storymap_helper.py

```diff
@@ -54,53 +64,313 @@ def __init__(
             ),
-            summary="A story map for Geodesignhub negotiation with project id {project_id}".format(
+            "{project_description}": "A story map for Geodesignhub negotiation with project id {project_id}".format(
                 project_id=self._gdh_design_details.gdh_design_details.project_id
             ),
-            by_line="Geodesignhub",
-            image=my_cover_image,
-        )
+            "{negotiated_design_item_id}": self._negotiated_design_item_id,
+        }
+
+        yaml_str = yaml.dump(self._storymap_template)
+        for placeholder, value in placeholders.items():
+            yaml_str = yaml_str.replace(placeholder, value)
+        self._storymap_template = yaml.safe_load(yaml_str)
 
     def create_new_storymap(self) -> StoryMap:
-        """Create a blank storymap and return the instance"""
+        """Create a blank StoryMap with the given title and return the instance."""
+        # Use the title from the template or fallback to a default
+
         my_geodesignhub_project_story = StoryMap()
         return my_geodesignhub_project_story
+    
+    def populate_storymap_from_template(self):
+        """Populate the StoryMap using the YAML template."""
+        # Set the cover
+        self._set_cover()
 
-    def add_project_text_gallery_to_storymap(self):
-        """Add some basic details about Geodesignhub project"""
-        mission_heading_text = Text(
-            text="Our Mission Statement", style=TextStyles.HEADING
-        )
-        mission_paragraph = Text(
-            text="Our mission description with more information about our goals"
+        # Iterate over the panels and add content accordingly
+        panels = self._storymap_template.get("panels", [])
+        for panel in panels:
+            panel_type = panel.get("type")
+            if panel_type == "text":
+                self._add_text(panel)
+            elif panel_type == "map":
+                self._add_map(panel)
+            elif panel_type == "image":
+                self._add_image(panel)
+            elif panel_type == "gallery":
+                self._add_gallery(panel)
+            elif panel_type == "separator":
+                self._add_separator()
+            elif panel_type == "table":
+                self._add_table(panel)
+            else:
+                logger.warning(f"Unsupported panel type: {panel_type}")
+
+    def _set_cover(self):
+        """Set the cover of the StoryMap."""
+        if self._storymap is None:
+            return  # Exit if StoryMap is not initialized
+
+        # Construct the cover directly using available template data and format strings
+        cover = self._storymap_template.get("cover", {})
+        title = cover.get("title", "Project Title")
+        summary = cover.get("subtitle", "Project Description")
+        cover_image_url = cover.get("cover_image_url", "https://story.maps.arcgis.com/sharing/rest/content/items/2fcc801c983a402eb427dd5cd07ee759/data")
```

Comment:
Nit: It would be nice to have a cover image, the current one the world map is not that great!

Reply:
e.g. look at the panel here: https://storymaps.arcgis.com/stories/fe0787354a79469d99a86c2335f2ab57 if we have a royalty free image like this would be best

Reply:
Found the royal free stock images